optaplanner-test/src/main/java/org/optaplanner/test/api/score/stream/AbstractAssertion.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.test.api.score.stream;
+
+public abstract class AbstractAssertion<Solution_, A extends AbstractAssertion<Solution_, A, V>, V
+        extends AbstractConstraintVerifier<Solution_, A, V>> {
```

Comment:
Mind renaming `V` to `Verifier_` and `A` to `Assertion_`?

Do we really need this Foo<F extends Foo<F>? I did that for Score and I regret it.
Do we really need types A and V?

Ok to see this merges already as is and discuss on master.